java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,323 @@
+     * </ul>
+     */
+    public interface ScoredRangeQuery extends RangeQuery {}
+
+    /**
+     * Represents a range by index (rank) in a sorted set.<br>
+     * The <code>start</code> and <code>stop</code> arguments represent zero-based indexes.
+     */
+    @Getter
+    public static class RangeByIndex implements ScoredRangeQuery {
+        private final String start;
+        private final String end;
+
+        /**
+         * Creates a range by index (rank) in a sorted set.<br>
+         * The <code>start</code> and <code>stop</code> arguments represent zero-based indexes.
+         *
+         * @param start The start index of the range.
+         * @param end The stop index of the range.
+         */
+        public RangeByIndex(long start, long end) {
+            this.start = Long.toString(start);
+            this.end = Long.toString(end);
+        }
+
+        @Override
+        public Limit getLimit() {
+            return null;
+        }
+    }
+
+    /**
+     * Represents a range by score in a sorted set.<br>
+     * The <code>start</code> and <code>stop</code> arguments represent score boundaries.
+     */
+    @Getter
+    public static class RangeByScore implements ScoredRangeQuery {
+        private final String start;
+        private final String end;
+        private final Limit limit;
+
+        /**
+         * Creates a range by score in a sorted set.<br>
+         * The <code>start</code> and <code>stop</code> arguments represent score boundaries.
+         *
+         * @param start The start score boundary.
+         * @param end The stop score boundary.
+         * @param limit The limit argument for a range query. Defaults to null. See <code>Limit</code>
+         *     class for more information.
+         */
+        public RangeByScore(
+                @NonNull RangeOptions.ScoreRange start,
+                @NonNull RangeOptions.ScoreRange end,
+                @NonNull Limit limit) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = limit;
+        }
+
+        /**
+         * Creates a range by score in a sorted set.<br>
+         * The <code>start</code> and <code>stop</code> arguments represent score boundaries.
+         *
+         * @param start The start score boundary.
+         * @param end The stop score boundary.
+         */
+        public RangeByScore(
+                @NonNull RangeOptions.ScoreRange start, @NonNull RangeOptions.ScoreRange end) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = null;
+        }
+    }
+
+    public static String[] createZrangeArgs(
+            String key, RangeQuery rangeQuery, boolean reverse, boolean withScores) {
+        String[] arguments = new String[] {key, rangeQuery.getStart(), rangeQuery.getEnd()};
+
+        if (rangeQuery instanceof RangeByScore) {
+            arguments = concatenateArrays(arguments, new String[] {"BYSCORE"});
```

Comment:
create static constants for these strings, so that we don't hardcode strings here and in test.